### PR TITLE
Specialize `isfinite` for half and bfloat16

### PR DESCRIPTION
**Pull request by crcrpar**

### CODE REVIEW

File: test/test_gpu3.cpp [hunk 1]
```diff
@@ -8603,6 +8604,33 @@ TEST_F(NVFuserTest, IntegerDivision_CUDA) {
...
+    auto fusion = fusion_ptr.get();
+    FusionGuard fg(fusion);
+    auto tv0 = makeContigTensor(1, nvfuser_dtype);
+    fusion->addInput(tv0);
+    auto tv1 = isfinite(tv0);
+    fusion->addOutput(tv1);
+
+    auto options = at::TensorOptions().dtype(aten_dtype).device(at::kCUDA, 0);
+    std::array<float, 3> data{1.0, INFINITY, NAN};
+    const auto input = at::from_blob(data.data(), {1}, {3}).to(options);
```
Comment (zasdfgbnm) [inline test/test_gpu3.cpp:8623]: Based on https://pytorch.org/cppdocs/api/function_namespacetorch_1ac009244049812a3efdf4605d19c5e79b.html, should this be
```suggestion
    const auto input = at::from_blob(data.data(), {3}, {1}).to(options);
```

**Review by zasdfgbnm — APPROVED** [review]
!build

**Comment by zasdfgbnm** [comment]
!build

File: runtime/bf16_support.cu [hunk 2]
```diff
@@ -20,6 +20,10 @@ struct __align__(2) __bfloat {
     __x = __float2bfloat(f).__x;
   }
 
+  __device__ uint16_t row() const {
```
Comment (zasdfgbnm) [inline runtime/bf16_support.cu:23]:
```suggestion
  __device__ uint16_t raw() const {
```
😅

Comment (crcrpar) [reply runtime/bf16_support.cu:23]: i am stupid...

**Comment by xwang233** [comment]
!build

**Comment by zasdfgbnm** [comment]
CI failure unrelated, I think it is https://github.com/NVIDIA/Fuser/issues/375

**Pull request merged**